At the moment hugin-app would still compile without proper C++11 support with a tiny patch
(but probably not for much longer).

--- src/hugin_base/hugin_utils/utils.h.orig
+++ src/hugin_base/hugin_utils/utils.h
@@ -43,6 +43,23 @@
 // on unix_like systems don't use CurrentTime, this is defined as a macro in X.h and breaks the debug messages
 // on windows we can't use GetCurrentTime because this is replaced with GetTickCount
 
+const // It is a const object...
+class nullptr_t
+{
+  public:
+    template<class T>
+    inline operator T*() const // convertible to any type of null non-member pointer...
+    { return 0; }
+
+    template<class C, class T>
+    inline operator T C::*() const   // or any type of null member pointer...
+    { return 0; }
+
+  private:
+    void operator&() const;  // Can't take address of nullptr
+
+} nullptr = {};
+
 #ifdef __GNUC__
     // the full function name is too long..
 //  #define DEBUG_HEADER hugin_utils::CurrentTime() <<" (" << __FILE__ << ":" << __LINE__ << ") "  << __PRETTY_FUNCTION__ << "()" << std::endl << "    "
--- src/hugin_base/lensdb/LensDB.cpp.orig
+++ src/hugin_base/lensdb/LensDB.cpp
@@ -844,7 +844,7 @@ public:
         {
             return false;
         };
-        std::ifstream input(filename);
+        std::ifstream input(filename.c_str());
         if (input.is_open())
         {
             while (!input.eof())
